savant/gstreamer/buffer_processor.py

```diff
@@ -0,0 +1,90 @@
+"""Buffer processor for GStreamer pipeline."""
+import logging
+from queue import Queue
+from typing import Iterator
+
+from gi.repository import Gst
+
+from savant.config.schema import PipelineElement
+from savant.utils.fps_meter import FPSMeter
+from savant.utils.sink_factories import SinkMessage
+
+
+class GstBufferProcessor:
+    def __init__(self, queue: Queue, fps_meter: FPSMeter):
+        """Buffer processor for GStreamer pipeline.
+
+        :param queue: Queue for output data.
+        :param fps_meter: FPS meter.
+        """
+
+        self._logger = logging.getLogger(
+            f'{self.__class__.__module__}.{self.__class__.__name__}'
+        )
+        self._queue = queue
+        self._fps_meter = fps_meter
+
+    # Buffer handlers
+    def prepare_input(self, buffer: Gst.Buffer):
```

Comment:
I suggest adding the @abstractmethod decorator to all methods that must necessarily be implemented.